### PR TITLE
TR-139: Avoid "context limit exceeded" error

### PR DESCRIPTION
**Pull request by binaryberserker**

### CODE REVIEW

File: public/react/Configure/components/Teams/ConfigureTeamsForm.tsx [hunk 1]
```diff
@@ -68,8 +70,13 @@ const ConfigureTeamsForm: FC<ConfigureTeamsFormProps> = ({
           onSave={update}
         />
         <InheritanceToggleField
-          isInheriting={!userData.spreadEffortAcrossDates && typeof userData.spreadEffortAcrossDates === "object"}
-          onInheritanceChange={(shouldCustomize) => toggleInheritance("spreadEffortAcrossDates", shouldCustomize)}
+          isInheriting={
+            // Look for undefined and null but not false
+            userData.spreadEffortAcrossDates == null
```
Comment (binaryberserker) [inline public/react/Configure/components/Teams/ConfigureTeamsForm.tsx:75]: this ended up being the fix for the spread null check

File: public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { type Configuration, createUpdatedTeamData } from "../team-configuration";
+
+export function sanitizeAllTeamData(
+  allTeamData: any,
```
Comment (DavidNic11) [inline public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts:4]: types plz

File: public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      return notEmpty(sanitizedConfig, [configKey, sanitizedConfig]);
+    });
+
+    return notEmpty(configs, [teamKey, configs]);
+  });
+
+  return sanitized;
+}
+
+function entriesFlatMap<T, R extends readonly [string, any]>(
```
Comment (DavidNic11) [inline public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts:31]: Can we name `T` and `R` something semantic

same below

File: public/react/Configure/components/Teams/services/team-configuration/hooks/useSaveAllTeamData.tsx [hunk 4]
```diff
@@ -101,20 +134,23 @@ export const useSaveTeamData: UseSaveTeamData = (config) => {
...
-          configuration: updates,
-        })
+      const sanitized = sanitizeAllTeamData(
+        allTeamData.userData,
+        teamName,
+        hierarchyLevel,
+        updates
       );
+
+      save(sanitized as any);
```
Comment (DavidNic11) [inline public/react/Configure/components/Teams/services/team-configuration/hooks/useSaveAllTeamData.tsx:153]: type plz

File: public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts (hunk 3 above)
Comment (binaryberserker) [reply public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts]: these funcs were just supposed to make the sanitize func cleaner and easier to read, figuring out what the types need to be for them is more of a hassle than I wanna fool with, I just inlined them in the sanitize func

**Review by DavidNic11 — APPROVED** [review]

**Pull request merged**